lib/utils/proxyconn.go

```diff
@@ -42,6 +42,14 @@ func (o CombinedReadWriteCloser) Write(p []byte) (int, error) {
 	return o.w.Write(p)
 }
 
+func (o CombinedReadWriteCloser) ReadFrom(r io.Reader) (n int64, err error) {
+	return io.Copy(o.w, r)
+}
+
+func (o CombinedReadWriteCloser) WriteTo(w io.Writer) (n int64, err error) {
+	return io.Copy(w, o.r)
+}
+
```

Comment:
I don't know how I feel about this, aren't `ReaderFrom` and `WriterTo` supposed to only be implemented on a type if the type is known to have good ways to implement the respective methods?